lib/internal/http.js

```diff
@@ -238,6 +239,87 @@ function filterEnvForProxies(env) {
+  'Cross-Origin-Resource-Policy', 'CTA-Common-Access-Token', 'DASL', 'Date', 'DAV', 'Default-Style', 'Delta-Base',
+  'Deprecation', 'Depth', 'Derived-From', 'Destination', 'Detached-JWS', 'Differential-ID', 'Dictionary-ID', 'Digest',
+  'DPoP', 'DPoP-Nonce', 'Early-Data', 'EDIINT-Features', 'ETag', 'Expect', 'Expect-CT', 'Expires', 'Ext', 'Forwarded',
+  'From', 'GetProfile', 'Hobareg', 'Host', 'HTTP2-Settings', 'If', 'If-Match', 'If-Modified-Since', 'If-None-Match',
+  'If-Range', 'If-Schedule-Tag-Match', 'If-Unmodified-Since', 'IM', 'Include-Referred-Token-Binding-ID', 'Isolation',
+  'Keep-Alive', 'Label', 'Last-Event-ID', 'Last-Modified', 'Link', 'Link-Template', 'Location', 'Lock-Token', 'Man',
+  'Max-Forwards', 'Memento-Datetime', 'Meter', 'Method-Check', 'Method-Check-Expires', 'MIME-Version', 'Negotiate',
+  'NEL', 'OData-EntityId', 'OData-Isolation', 'OData-MaxVersion', 'OData-Version', 'Opt', 'Optional-WWW-Authenticate',
+  'Ordering-Type', 'Origin', 'Origin-Agent-Cluster', 'OSCORE', 'OSLC-Core-Version', 'Overwrite', 'P3P', 'PEP',
+  'PEP-Info', 'Permissions-Policy', 'PICS-Label', 'Ping-From', 'Ping-To', 'Position', 'Pragma', 'Prefer',
+  'Preference-Applied', 'Priority', 'ProfileObject', 'Protocol', 'Protocol-Info', 'Protocol-Query',
+  'Protocol-Request', 'Proxy-Authenticate', 'Proxy-Authentication-Info', 'Proxy-Authorization', 'Proxy-Features',
+  'Proxy-Instruction', 'Proxy-Status', 'Public', 'Public-Key-Pins', 'Public-Key-Pins-Report-Only', 'Range',
+  'Redirect-Ref', 'Referer', 'Referer-Root', 'Referrer-Policy', 'Refresh', 'Repeatability-Client-ID',
+  'Repeatability-First-Sent', 'Repeatability-Request-ID', 'Repeatability-Result', 'Replay-Nonce',
+  'Reporting-Endpoints', 'Repr-Digest', 'Retry-After', 'Safe', 'Schedule-Reply', 'Schedule-Tag', 'Sec-Fetch-Dest',
+  'Sec-Fetch-Mode', 'Sec-Fetch-Site', 'Sec-Fetch-Storage-Access', 'Sec-Fetch-User', 'Sec-GPC', 'Sec-Purpose',
+  'Sec-Token-Binding', 'Sec-WebSocket-Accept', 'Sec-WebSocket-Extensions', 'Sec-WebSocket-Key',
+  'Sec-WebSocket-Protocol', 'Sec-WebSocket-Version', 'Security-Scheme', 'Server', 'Server-Timing', 'Set-Cookie',
+  'Set-Cookie2', 'SetProfile', 'Signature', 'Signature-Input', 'SLUG', 'SoapAction', 'Status-URI',
+  'Strict-Transport-Security', 'Sunset', 'Surrogate-Capability', 'Surrogate-Control', 'TCN', 'TE', 'Timeout',
+  'Timing-Allow-Origin', 'Topic', 'Traceparent', 'Tracestate', 'Trailer', 'Transfer-Encoding', 'TTL', 'Upgrade',
+  'Urgency', 'URI', 'Use-As-Dictionary', 'User-Agent', 'Variant-Vary', 'Vary', 'Via', 'Want-Content-Digest',
+  'Want-Digest', 'Want-Repr-Digest', 'Warning', 'WWW-Authenticate', 'X-Content-Type-Options', 'X-Frame-Options',
+];
+
+
+const IANAHTTPHeaders = new SafeMap([
+  ...IANARegisteredHeaders.map((fieldname) => ([fieldname, fieldname.toLowerCase()])),
+  ...IANARegisteredHeaders.map((fieldname) => ([fieldname.toLowerCase(), fieldname.toLowerCase()])),
```

Comment:
@joyeecheung would doing this work on a startup slow us down somewhat? Maybe it's better to just have a massive file with all of those?